capsules/src/ble_advertising_driver.rs

```diff
@@ -489,49 +380,41 @@ impl App {
                 slice.as_mut()[PACKET_HDR_PDU] = BLEAdvertisementType::NonConnectUndirected as u8;
                 ReturnCode::SUCCESS
             })
-            .unwrap_or_else(|| ReturnCode::ESIZE)
+            .unwrap_or(ReturnCode::FAIL)
     }
 
-    fn set_gap_data(&mut self, gap_type: BLEGapType) -> ReturnCode {
+    // Replace AdvData for `ConnectUndirected`, `NonConnectUndirected` and `ScanUndirected`
+    fn replace_adv_data(&mut self) -> ReturnCode {
         self.app_write
             .take()
             .as_ref()
             .map(|slice| {
-                let idx = self.idx;
-                let end = idx + slice.len() + 2;
-
-                if end <= PACKET_LENGTH {
-                    let result = self.advertisement_buf
+                let payload_length = slice.len() as u8;
+                if payload_length >= MIN_PAYLOAD_LEN && payload_length <= MAX_PAYLOAD_LEN {
```

Comment:
the buffer length should be 3 => x => 31 

Recall, a gap field has `Length | Gap Type | Data`

However, these are redundant checks and re aperformed by `libtock`